Add REST generator documentation (#42)
SAP/cloud-sdk: docs/js/features/openapi/overview.mdx

---
title: OpenApi with the Cloud SDK for JavaScript
sidebar_label: Overview
description: This article describes how generate a client from the a service definition and how to use the client.
keywords:
- sap
- cloud
- sdk
- OpenApi
- rest
- connectivity
- JavaScript
- TypeScript
---

import useBaseUrl from '@docusaurus/useBaseUrl'

## What is the OpenApi generator and what does it offer for you

The OpenApi generator is a command line tool to generate clients from an [OpenApi specification](https://swagger.io/specification/).

> OpenApi specifications are the defacto standard to define RESTful webservice in a language-agnostic way comparable to the `edmx` service definition for [OData clients](../odata/overview.mdx).

Reviewer: ```suggestion
OpenApi specifications are the defacto standard to define RESTful webservices in a language-agnostic way comparable to the `edmx` service definition for [OData clients](../odata/overview.mdx).
```

> As you would expect there exists already a [openapi-generator](https://github.com/OpenAPITools/openapi-generator) which offers client generator for a big number of languages.

Reviewer: ```suggestion
As you would expect an [openapi-generator](https://github.com/OpenAPITools/openapi-generator) which offers client generation for a big number of languages already exists.
```

So why do we provide an extra generator in the SDK?

> The reason we provided an extra generator based on the open source one is that we wanted to include the connectivity layer of the SDK also for REST APIs and not only for OData.

Reviewer: ```suggestion
The reason we provided an extra generator based on the existing one is that we wanted to include the connectivity layer of the SDK for other REST APIs as well and not only for OData.
```
